### ISSUE TITLE
Support "exists" when it gets pushed down to storage

### ISSUE THREAD

**Issue by affo** [influxdata/influxdb#15251]
Query:

```
from(bucket: "system_usage")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r._measurement == "http_request" and exists r.org_id)
  |> filter(fn: (r) => r._field == "resp_bytes")
  |> filter(fn: (r) => r.endpoint == "/api/v2/write")
  |> tableFind(fn: (key) => true)
```

Result:

```
error calling function "tableFind": left hand side: left hand side: right hand side: unsupported semantic expression type *semantic.UnaryExpression
```

I was able to see that thank to `tableFind`, otherwise I could only see `internal error`.

It seems that storage does not support `exists`.

The error happens here: https://github.com/influxdata/influxdb/blob/2909927ab0eaa2943905f5beab5bbb8d71ca18b8/storage/reads/predicate.go#L295

**Comment by jsternberg** [comment]
An interesting part of this is that there is code that converts the `exists` call to the appropriate predicate. So we need to check why that code isn't working.

**Comment by jsternberg** [comment]
I don't know why I never connected this to this fix: #15557.

This should be fixed now.
